src/extensions/dead-clicks-autocapture.ts

```diff
@@ -0,0 +1,91 @@
+import { PostHog } from '../posthog-core'
+import { DEAD_CLICKS_ENABLED_SERVER_SIDE } from '../constants'
+import { isBoolean, isObject } from '../utils/type-utils'
+import { assignableWindow, document, LazyLoadedDeadClicksAutocapture } from '../utils/globals'
+import { logger } from '../utils/logger'
+import { DecideResponse } from '../types'
+
+const LOGGER_PREFIX = '[Dead Clicks]'
+
+export class DeadClicksAutocapture {
```

Comment:
am also thinking about https://github.com/PostHog/posthog-js/pull/1439 when making this structure.
we have almost exactly the same logic repeated for every lazy loaded extension

i'm interested in extending this simple wrapper on lazy loaded extension to replay (would save up to 13% of bundle)  and then maybe abstracting it so every lazy loaded extension could be added the same way